mirdata/download_utils.py

```diff
@@ -56,9 +57,12 @@ def downloader(
         remotes (dict or None):
             A dictionary of RemoteFileMetadata tuples of data in zip format.
             If None, there is no data to download
+        index (core.Index):
+            A mirdata Index class, which may contain a remote index to be downloaded
+            or a subset of remotes to download by default.
         partial_download (list or None):
             A list of keys to partially download the remote objects of the download dict.
-            If None, all data is downloaded
+            If None, all data specified by the index is downloaded
         info_message (str or None):
```

Comment:
With this phrasing I understand that the index contains the `remote index` (if any) AND the `remote data` to download, but that's not the case right?

Reply:
I created a new `core.Index` class, which is just a container of information for indexes. It has three important attributes:
```
* remote : None (default) or a RemoteFileMetadata object, saying where to download the index from
* partial_download : None (default) or a list of keys of `REMOTES`, in case the index only requires a subset of the data.
* get_path(): returns the path to the index. The function looks to see if remote is None or not to know if the indexes in the usual spot, or somewhere else.
```

So all that to answer your question - the index might contain a remote index (which should get downloaded with the rest of the data), and might also specify a subset of remote data. I was imagining e.g. a dataset with two versions of annotations - if each version is a separate download, you might want to only download the version of the annotations you're using for that version, rather than everything.

Reply:
I see, that's great! If nothing is specified in remotes, does `mirdata` download all versions?

Reply:
dismiss, you already answer that offline in the meeting